### PR TITLE
PBAB Support for AB Disc + Refactor

### PR DESCRIPTION
**Pull request by grantoesterling**
## What's New
- We are adding a 'block-talk' type channel for PBAB, so adding support for AB alertbot to sent new PBAB mints (for all PBAB projects) to a new webhook (see https://app.asana.com/0/1201517173861745/1202292292892976/f)
- Added GraphQL query-er to get all PBAB contracts
- Refactored code from last few PRs so we aren't relying on janky env variable stuff as much - I like this better now :) 
-- As a result, the contract of the event gets passed around to more functions
- AB won't tweet PBAB mints, but it will send them to the new `PBAB_DISCORD_WEBHOOK_URL` (might consider renaming to avoid confusion for PBAB projects forking this)

This PR ended up being a bit bigger than I was anticipating, so sorry for that!! Again, a majority of the changes here are formatting things. After this PR I'll open one to move this repo over to using prettier (just did that for artbot) to prevent this from happening again in the future

## TODO Before merge
- Create and add new webhook!

### CODE REVIEW

**Review by Jbern16 — APPROVED** [review]
Looks awesome @grantoesterling,  I believe this streamlines PBAB partner setup quite a bit, one request would be to update docs accordingly

File: src/api_parser.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  fetchOptions: () => ({
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  }),
+});
+
+const _getPBABContracts = gql`
+  query getPBABContracts($ids: [ID]!) {
+    contracts(where: { id_not_in: $ids }) {
```
Comment (ryley-o) [inline src/api_parser.ts:19]: naming nit: Can we either rename this to be something generic like `_getContracts` with arg `$ids_not_in`, or maybe keep the function name, remove the args, and just automatically include `${AB_CONTRACTS}` in the gql string?

**Review by ryley-o — APPROVED** [review]
LGTM, TYVM @grantoesterling!

Minor nit on gql query, happy to chat if you want to discuss more 👍

File: src/api_parser.ts (hunk 1 above)
Comment (grantoesterling) [reply src/api_parser.ts]: done!

**Pull request merged**